src/gengowatcher/secure_storage.py

```
@@ -0,0 +1,135 @@
+import logging
+from cryptography.fernet import Fernet
+from cryptography.hazmat.primitives import hashes
+from cryptography.hazmat.primitives.kdf.pbkdf2 import PBKDF2HMAC
+from pathlib import Path
+from typing import Optional
+
+
+class SecureKeyStorage:
+    """Securely stores and retrieves API keys using encryption"""
+    
+    def __init__(self, storage_file: str = "captcha_keys.json", logger: logging.Logger = None):
```

Ruff (0.12.2)

PEP 484 prohibits implicit `Optional`

Convert to `Optional[T]`

(RUF013)